### ISSUE TITLE
More maintainers for vimwiki

### ISSUE THREAD

**Issue by Nudin** [vimwiki/vimwiki#624]
Vimwiki currently only has very little maintainers. PRs are waiting very long until they get merged or commented. This causes frustration and delays the project.

New maintainers are appointed by @EinfachToll. Use this Issue if you want to apply for maintainerrights.

This issue superseeds the older issue #402.

**Comment by ranebrown** [comment]
I'm interested in being a maintainer. Let me know if you need any other info. Thanks!

**Comment by hq6** [comment]
I am also interested in being a maintainer / developer. I have some experience with `vimscript` and am excited about adding new features. Additionally, I have enough reputation on StackOverflow to create a tag there and redirect questions to that community is it is desirable.

**Comment by EinfachToll** [comment]
I will add you both as participants. I also added a few advice to CONTRIBUTING.md for you. Hopefully you are aware of your huge responsibility and not only try to add new fancy features but fix old and boring bugs ;)

P.S. Just so you know, when I'm sometimes inactive for a few weeks, that doesn't mean anything. My contributions come slowly, but I don't see me stopping or growing tired or something.

**Comment by EinfachToll** [comment]
OK, I think I've sent you invitations to be collaborators. Let me know if you need any rights you don't have. I don't completely understand the rights management here on Github.

**Comment by hq6** [comment]
@EinfachToll, I do not find `CONTRIBUTING.md` in the repository. Can you offer a link?

**Comment by Nudin** [comment]
@hq6 it's in the [dev-branch](https://github.com/vimwiki/vimwiki/blob/dev/CONTRIBUTING.md)!

**Comment by Nudin** [comment]
@EinfachToll could you please add me to the `vimwiki` Github-group? So that I can create new repositories. (Beside others for further documentation and testing)

**Comment by hq6** [comment]
If we're going to have separate but related repositories, it may be more
efficient if you add me and Rane to well.

Otherwise, we'd have to be manually added as collaborators on each repo
after creation.

**Comment by thecliguy** [comment]
Just in case any maintainers don't know, there is a vimwiki IRC channel. I'd encourage you to join, details below:

`#vimwiki` on Freenode ([webchat](https://webchat.freenode.net/?channels=#vimwiki), also synced to Matrix/Riot: `#freenode_#vimwiki:matrix.org`)

**Comment by brennen** [comment]
Howdy all - I'd be happy to at least help triage issues and such from time to time...

**Comment by tinmarino** [comment]
@EinfachToll : I'd also like to be maintainer. To edit developers pull request, rebase my test branch incrementally when stable without harassing current few maintainers (i.e ranebrown). And to add some workforce for issue fixing.

**Comment by ranebrown** [comment]
@Nudin do you have the ability to add maintainers?

**Comment by Nudin** [comment]
> @Nudin do you have the ability to add maintainers?

Yes, I can. I added @tinmarino and @brennen. At the same time I protected the `master` branch so that only @EinfachToll @ranebrown and me can push to that branch – that allows to hand out rights more relaxed.

**Comment by tinmarino** [comment]
@Nudin It seems I am not collaborator (I cannot add tags nor accept PR on dev)

**Comment by guntari** [comment]
I've been using vimwiki work-daily I'd say about 7, 8 or more years now. Ever since CamelCase was the default for links. I actually was so fond of camel case and so distressed by its loss that  I forked your code for my own purposes a long while back. I'm interested in trying my hand at maintainer.  I absolutely love vimwiki, it has been an invaluable daily core tool for me. I've been a vi and vim user since the early 90s.

**Comment by Nudin** [comment]
@tinmarino I added you on Feb 1st. But you still haven't "accepted the invite". You should have gotten an email or similar with a link to accept the invite.
You can also try this link: https://github.com/vimwiki/vimwiki/invitations

**Comment by tinmarino** [comment]
@Nudin, I just did not see it, it expired. Could you invite me again, so I don't have to make PR, sorry for the annoyance ?

> Pending invitations will expire after 7 days, restoring any unclaimed licenses.

**Comment by Nudin** [comment]
I just gave a few people further rights. I'm happy to see that we now have a few active maintainers that can & will merge PR.

- I will give the right to "triage issues" (add labels, projects, close, etc) to everyone who is interested – just drop a note.
- I'll hand out write access for the dev-branch to people who already made a few commits.
- If someone wants to work on feature branches in this repo, let me know.

**Comment by chipsenkbeil** [comment]
I'm up for helping maintain this plugin. Some examples of direct and indirect contributions thus far:
- writing up a [formal specification](https://github.com/vimwiki/vimwiki/pull/1025) for the language
- adding [multi-line comments](https://github.com/vimwiki/vimwiki/pull/967)
- building a Rust library to parse and transform vimwiki files ([vimwiki-rs](https://github.com/chipsenkbeil/vimwiki-rs))
- expanding vimwiki's presence by adding it to [cobalt](https://github.com/cobalt-org/cobalt.rs/issues/817) and [zola](https://github.com/getzola/zola/issues/1276)

**Comment by Nudin** [comment]
@chipsenkbeil: Added you to "triage", will add you to "write" soon.

I also started cleaning up the numerous rights, members and teams. It was (and still is) quite messy and there are a lot of accounts which haven't been active in years. Seems like an unnecessary security risk, especially when considering how many people use this project and that most users don't use the last release but the current master…

**Comment by Nudin** [comment]
@vimwiki/admins @vimwiki/maintainer @vimwiki/contributors @vimwiki/former-contributors @brennen @hq6 @guntari 
If anyone wonders about messages of being added/removed from some repos/teams/rights: 

1. I am cleaning up the permissions model, so that peoples don't have individual rights but get them though being part of a team of the organization.
2. People who had write-access and didn't contribute for several years ('former-contributors') now only have triage rights. This is not meant personal nor judging but for security reasons. If you come back you can get write access again. :)

**Comment by chipsenkbeil** [comment]
@Nudin so the primary invite should be for the vimwiki organization? I got two: the organization and the vimwiki/vimwiki repo. The former worked and I accepted whereas the latter gives me an error about it not being available. Either way, I can't add tags to issues or do anything else with them despite being in the vimwiki organization, so not sure what permission triage rights should give me. :)

**Comment by brennen** [comment]
> I am cleaning up the permissions model, so that peoples don't have individual rights but get them though being part of a team of the organization.

Sounds like a fine idea.

**Comment by Nudin** [comment]
@chipsenkbeil Yes, that's right you should be part of the organization and within that in one or more groups. Something wasn't set up correctly, you should now have triage and write access – except for the `master` branch.

**Comment by DannyDannyDanny** [comment]
If it's still relevant, I'd like to apply as a maintainer. I can help triage PRs and reproduce bugs. I'm still inexperienced with vim script but I'm experimenting and - one day - I hope to contribute functionality to the project.

**Comment by agarie** [comment]
I've been a vimwiki user for some years now---I moved my notes to a wiki and (re-)started a diary back in 2019. I recently decided to check on some issues that I had and eventually learned that the `dev` branch has fixes for all of them. After that I noticed this issue and #1235, and guess I should give it a try.

Thus I'd like to apply as a maintainer. Should I contact anybody in private? Thank you!

**Comment by tinmarino** [comment]
Hello @agarie,

You can create some some issues and PR at any time. Before being maintainer, it would be nice to familiarize yourself with the code. Please start with implementing 1 or 2 [good first issues](https://github.com/vimwiki/vimwiki/labels/good%20first%20issue) and commenting some of the current PR in order to foresee what you are applying for and to introduce yourself, your code style to the team.

Cheers.

**Comment by tinmarino** [comment]
Hello @Nudin @brennen,

Can you please raise my rights to contribute to #1235 (continuous release).
More specifically I would like to focus on the automatic tests triggered by the  [Vimwiki GitHub Action](https://github.com/vimwiki/vimwiki/actions/runs/4388119096) currently not working due to access rights.

Note that it seems that the former travis CI is now paid, so it has been replaced by GitHub Actions (yes I come late lol). This should not represent serious changes weather improvements or drawback, just a change of responsible company.

Saludos desde Chile.

**Comment by Nudin** [comment]
@tinmarino I raised your rights. I would however recommend you to set up 2FA for your account.

**Comment by tinmarino** [comment]
Thank you @Nudin, that was fast! I configured my github 2FA as advised.

I agree that it is wiser due to the facts that vimwiki has many users and can execute anything on their computer.
